### ISSUE TITLE
Additional functionality

### ISSUE THREAD

**Issue by d-e-s-o** [d-e-s-o/nitrocli#40]
Having just gone through the process of getting a new Nitrokey set up I feel that at least the following functions also need to be supported:

- [ ] changing of the read-write state of the unencrypted volume
- [ ] setting of the user & admin PIN
- [ ] enabling of firmware upgrade mode

It is not quite clear to me whether the user & admin PIN could be set through GnuPG (which may alleviate the need for the second item), we may want to try that out

**Comment by d-e-s-o** [comment]
Ah, I forgot that Robin already has a change out for the PIN changing functionality :-)

**Comment by robinkrahl** [comment]
Valid points.  The functionality is provided by libnitrokey but not yet implemented in the nitrokey crate.  [Here](https://git.ireas.org/nitrokey-rs/tree/TODO.md) is a list of the libnitrokey functions that are not yet added to nitrokey.

As far as I see, the implementation in the nitrokey crate should be trivial.  I’ll include it in one of the next releases.

**Comment by jans23** [comment]
Yes, user & admin PIN could be set through GnuPG. Because not all users wants/need to use GPG, we added this important functionality to libnitrokey. I think for the same reason it would be great to have it in nitrocli.
-- 
Sent from a mobile device.

**Comment by robinkrahl** [comment]
> enabling of firmware upgrade mode

Supported in `nitrokey` 0.3.0.

**Comment by robinkrahl** [comment]
> hidden volumes
Supported in `nitrokey` 0.3.1.
